event_sidecar/src/event_stream_server.rs

```diff
@@ -68,7 +68,11 @@ pub(crate) struct EventStreamServer {
 }
 
 impl EventStreamServer {
-    pub(crate) fn new(config: Config, storage_path: PathBuf) -> Result<Self, ListeningError> {
+    pub(crate) fn new(
+        config: Config,
+        storage_path: PathBuf,
+        enable_legacy_filters: bool,
```

Comment:
A nit, but I'd rather avoid a bool flag here, as I can imagine that our "standard" filters will become legacy and the legacy ones will become "legacier" :)

How about an `enum Filters { V1_x, V2_x }` or something along these lines?